fintual-oss/paraffin-front
# Graph on cycle's page and alert when learning unit path is not completed

# Base PR
- PR based on main.

# Context
- Add a Learning Unit's section to the cycle page, where you can check a list of the learning units or the graph.
- We are adding an alert when a user clicks a learning unit node in the graph, for which she hasn't yet completed its path. For now, it will always be not completed as the data is not included in the endpoint, waiting for a PR in the backend that adds in the cycle-learning-units endpoint if each learning unit is completed by the user or not, so the alert is always shown.
- There is also a change in the theme of the graph's nodes.
- Refactor of the components within the Cycle Section.

# Changelog
- Removed repeated endpoint in endpoints.js
- Include LearningUnitsSection component in the Cycle Section
- LearningUnitsSection displays both list and graph tabs
- Created the component for the list tab
- Cycle page handles the learning unit redirection when clicking from the graph
- Created the alert on the learning unit page based on a query param
- Create the logic within LearningUnitsGraph component to check whether the path is completed or not for a specific learning unit
- Added theme to the graph
- Refactor of components folders, mainly moving earlier "learning-units-section" components inside "cycle-section" folder
- Renamed "GraphSection" component to "LearningUnitsGraph", as it handles the logic of the graph, applied to learning units case

# QA
- Run frontend and backend servers
- Go to http://localhost:3001/cycles/1
- Explore the page, hover on the graph nodes and edges
- Click on graph nodes and check both the url (includes "?not-completed" query param) and the alert shown in the learning unit page
